MeshGeoToolsLib/GeoMapper.cpp

```diff
@@ -115,34 +138,54 @@ void GeoMapper::mapData()
 		min_val = bounding_box.getMinPoint()[2];
 		max_val = bounding_box.getMaxPoint()[2];
 	}
-	std::size_t nPoints (points->size());
 
-	if (!is_borehole)
+	std::vector<GeoLib::Point*> const* points (_geo_objects.getPointVec(_geo_name));
+	std::size_t nPoints (points->size());
+	for (unsigned j=0; j<nPoints; ++j)
 	{
-		for (unsigned j=0; j<nPoints; ++j)
+		GeoLib::Point* pnt ((*points)[j]);
+		double offset = (_grid) ? (getMeshElevation((*pnt)[0],(*pnt)[1], min_val, max_val) - (*pnt)[2])
+			                    :  getDemElevation(*pnt);
+
+		if (!GeoLib::isBorehole(pnt))
+			continue;
+		GeoLib::StationBorehole* borehole = static_cast<GeoLib::StationBorehole*>(pnt);
+		const std::vector<GeoLib::Point*> layers = borehole->getProfile();
+		std::size_t nLayers = layers.size();
+		for (unsigned k=0; k<nLayers; ++k)
 		{
-			GeoLib::Point* pnt ((*points)[j]);
-			(*pnt)[2] = (_grid) ? this->getMeshElevation((*pnt)[0],(*pnt)[1], min_val, max_val)
-				                : this->getDemElevation(*pnt);
+			GeoLib::Point* layer_pnt = layers[k];
+			(*layer_pnt)[2] = (*layer_pnt)[2] + offset;
 		}
 	}
-	else
+}
+
+void GeoMapper::mapPointDataToDEM()
+{
+	std::vector<GeoLib::Point*> const& pnts(*_geo_objects.getPointVec(_geo_name));
+
+	for (auto pnt : pnts)
 	{
-		for (unsigned j=0; j<nPoints; ++j)
-		{
-			GeoLib::Point* pnt ((*points)[j]);
-			double offset = (_grid) ? (this->getMeshElevation((*pnt)[0],(*pnt)[1], min_val, max_val) - (*pnt)[2])
-				                    :  this->getDemElevation(*pnt);
-
-			GeoLib::StationBorehole* borehole = static_cast<GeoLib::StationBorehole*>(pnt);
-			const std::vector<GeoLib::Point*> layers = borehole->getProfile();
-			std::size_t nLayers = layers.size();
-			for (unsigned k=0; k<nLayers; ++k)
-			{
-				GeoLib::Point* layer_pnt = layers[k];
-				(*layer_pnt)[2] = (*layer_pnt)[2] + offset;
-			}
-		}
+		GeoLib::Point &p(*pnt);
+		p[2] = getDemElevation(p);
+	}
+}
+
+void GeoMapper::mapPointDataToMeshSurface(std::vector<GeoLib::Point*> const& pnts)
+{
+	GeoLib::AABB const aabb(
+		_surface_mesh->getNodes().cbegin(), _surface_mesh->getNodes().cend());
+	double const min_val(aabb.getMinPoint()[2]);
+	double const max_val(aabb.getMaxPoint()[2]);
+
+	for (auto pnt : pnts) {
+		GeoLib::Point &p(*pnt);
+		if (p[0] < aabb.getMinPoint()[0] || aabb.getMaxPoint()[0] < p[0])
```

Comment:
Could you add a comment like
`// checking if pnt is inside the bounding box of the _surface_mesh projected onto the x-y plane`
I guess that that's what is checked